src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,63 @@
+import classNames from 'classnames';
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useAppDispatch } from '../../app/hooks';
+import { setAppliedQuery } from '../../features/querySlice';
+import { SearchIconColor } from '../../types/SearchIconColor';
+import Icon from '../Icon/Icon';
+import styles from './SearchBar.module.scss';
+
+type Props = {
+  query: string;
+  setQuery: (value: string) => void;
+  applyQuery?: (value: string) => void;
+  searchIconColor: SearchIconColor;
+};
+
+export const SearchBar: React.FC<Props> = ({
+  query,
+  setQuery,
+  applyQuery,
+  searchIconColor,
+}) => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const onHandleQueryInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+
+    if (applyQuery) {
+      applyQuery(event.target.value);
```

Comment:
maybe add a primitive validation like trim method to delete user unnecessary spaces